src/components/Alert/index.js

```diff
@@ -1,50 +1,111 @@
+import { css, useTheme } from '@emotion/react'
 import styled from '@emotion/styled'
 import PropTypes from 'prop-types'
 import React from 'react'
-import Icon, { icons } from '../Icon'
+import Box from '../Box'
+import Icon from '../Icon'
 import Typography from '../Typography'
 
-const variantIcons = {
-  info: 'information-outline',
-  success: 'check',
-  warning: 'alert',
+const variants = {
+  beta: ({ theme }) => css`
+    background-color: ${theme.colors.serenade};
+    color: ${theme.colors.beta};
+  `,
+  info: ({ theme }) => css`
+    background-color: ${theme.colors.zumthor};
+    color: ${theme.colors.info};
+  `,
+  success: ({ theme }) => css`
+    background-color: ${theme.colors.foam};
+    color: ${theme.colors.success};
+  `,
+  warning: ({ theme }) => css`
+    background-color: ${theme.colors.pippin};
+    color: ${theme.colors.warning};
+  `,
 }
 
-const StyledAlert = styled(Typography)`
+export const AlertVariants = Object.keys(variants)
+
+const variantStyles = ({ variant, ...props }) => variants[variant]?.(props)
+
+const StyledContainer = styled(Box, {
+  shouldForwardProp: prop => !['variant', 'hasBackground'].includes(prop),
+})`
   display: flex;
+  flex-direction: row;
   align-items: center;
-  margin-bottom: ${({ theme }) => theme.space['2']};
-  font-weight: 700;
+  border-radius: ${({ theme }) => theme.radii.default};
+  color: ${({ theme }) => theme.colors.white};
+  padding: ${({ hasBackground }) => (hasBackground ? '12px' : '0')};
+  ${variantStyles}
+  ${({ hasBackground }) => !hasBackground && 'background-color: transparent'}
+`
+
+const StyledBox = styled(Box)`
+  display: flex;
+  flex-direction: column;
 `
 
-export const alertVariants = Object.keys(variantIcons)
-
-const Alert = ({ variant, icon, children, iconSize, ...props }) => (
-  <StyledAlert variant="default" role="alert" color={variant} {...props}>
-    <Icon
-      mr={2}
-      name={icon || variantIcons[variant]}
-      size={iconSize}
-      color={variant}
-    />
-    <span>{children}</span>
-  </StyledAlert>
+const Title = ({ color, text }) => (
+  <Typography variant="description" color={color} ml={2}>
+    {text}
+  </Typography>
 )
 
+const Alert = ({
+  variant,
+  children,
+  iconSize,
+  icon,
+  title,
+  hasBackground,
+  ...props
+}) => {
+  const theme = useTheme()
+  const color = theme.colors[variant]
+
+  return (
+    <StyledContainer variant={variant} hasBackground={hasBackground} {...props}>
```

Comment:
hasBackground is too much, we should add a background based on variant otherwise the user should pass `background="transparent"` to drop it no ?

Reply:
Before it was done with a variant but I think we cannot put every single design change into variant properties. 

So there are two options here:
- We add `background="transparent"` prop
- We add `type` prop, it will define if its `info`, `warning`, etc. and we change `variant` prop to define what design the alert is gonna be, like `outlined`, `filled`, `transparent`, etc. This option is a bit more pain in the ass but it makes a way more consistent component with logical properties.
In this option it will look like:
```js
<Alert type="warning" variant="transparent">...</Alert>
```

@DorianMaliszewski @chambo-e @philibea

Reply:
@DorianMaliszewski done!